rust/test_service_elections/src/lib.rs

```diff
@@ -18,20 +18,61 @@ mod service {
         pub winner: AccountNumber,
     }
 
+    impl ElectionRecord {
+        // #[primary_key]
+        fn by_id(&self) -> u32 {
+            self.id
+        }
+
+        // #[secondary_key(1)]
+        fn by_dates_key(&self) -> (TimePointSec, TimePointSec, u32) {
+            (self.voting_start_date, self.voting_end_date, self.id)
+        }
+    }
+
+    // TODO: Auto generated from above #[primary_key] and #[secondary_key(1)]
     impl TableRecord for ElectionRecord {
         type PrimaryKey = u32;
 
         fn get_primary_key(&self) -> Self::PrimaryKey {
-            self.id
+            self.by_id()
+        }
+
+        fn get_secondary_keys(&self) -> Vec<RawKey> {
+            vec![RawKey::new(self.by_dates_key().to_key())]
         }
     }
 
-    type ElectionsTable = Table<ElectionRecord>;
+    struct ElectionsTable(Table);
+    impl TableBase for ElectionsTable {
+        fn prefix(&self) -> Vec<u8> {
+            self.0.prefix.clone()
+        }
 
-    impl TableHandler<ElectionRecord> for ElectionsTable {
+        fn db_id(&self) -> DbId {
+            self.0.db_id
+        }
+    }
```

Comment:
What I did not like was these repetitions of:

```rust
    impl TableBase for XTable {
        fn prefix(&self) -> Vec<u8> {
            self.0.prefix.clone()
        }

        fn db_id(&self) -> DbId {
            self.0.db_id
        }
    }
```

It will always be the same... Any suggestions?